src/networking.c

```diff
@@ -2115,6 +2140,20 @@ char *getClientPeerId(client *c) {
     return c->peerid;
 }
 
+/* This function returns the client bound socket name, by creating and caching
+ * it if client->sockname is NULL, otherwise returning the cached value.
+ * The Socket Name never changes during the life of the client, however it
+ * is expensive to compute. */
+char *getClientSockname(client *c) {
+    char sockname[NET_PEER_ID_LEN];
+
+    if (c->sockname == NULL) {
+        genClientSockName(c,sockname,sizeof(sockname));
+        c->sockname = sdsnew(sockname);
+    }
+    return c->sockname;
+}
+
```

Comment:
maybe it worth in some way to find a way to re-use the code of the function above it?
same about genClientSockName, a lot of boilerplate code that's exactly the same, and tons of identical comments.

Reply:
I agree. I'll try to see if I can do something about it...

Reply:
I did a cleanup and removed lots of dup code. You can re-review.